include/unifex/reduce_stream.hpp

```diff
@@ -38,18 +38,33 @@
 
 namespace unifex {
 namespace _reduce {
-template <typename Operation>
+template <typename StreamSender, typename State, typename ReducerFunc, typename Receiver>
+struct _op {
+  struct type;
+};
+template <typename StreamSender, typename State, typename ReducerFunc, typename Receiver>
+using operation =
+    typename _op<
+        remove_cvref_t<StreamSender>,
+        remove_cvref_t<State>,
+        remove_cvref_t<ReducerFunc>,
+        remove_cvref_t<Receiver>>::type;
```

Comment:
I've recently been trying to move away from putting transformations like remove_cv_ref_t in the type-alias here as I find myself constantly wondering at the usage site of these sort of aliases whether it's forgotten to apply remove_cvref_t or not.

I would like to be consistent in our usage pattern for this kind of class as we use this ADL-isolation pattern everywhere. We can tackle that in another PR though.

Reply:
Yes, I know I had promised to fix all that, but I've never had time. :-(